apps/api/src/server/repositories/case.repository.js

```diff
@@ -411,11 +411,13 @@ export const createNewStatuses = (id, status) => {
  *
  * @param {number} id
  * @param {{status: string | object, data: {regionNames?: string[]}, currentStatuses: object[], setReference: boolean}} updateData
+ * @param {Promise<any>[]} createFolders
  * @returns {any}
  */
 export const updateApplicationStatusAndDataById = (
 	id,
-	{ status, data, currentStatuses, setReference = false }
+	{ status, data, currentStatuses, setReference = false },
+	/** @type {any[] | Promise<any>} */ createFolders
```

Comment:
I would not call this `createFolders`, but something a bit more generic. We could in theory pass any other transaction that we might want here.